mteb/tasks/BitextMining/multilingual/BibleNLPBitextMining.py

```diff
@@ -0,0 +1,955 @@
+    hf_lang_subset2isolang = {}
+    for x in _LANGUAGES:
+        if "-" not in x:
+            for y in _LANGUAGES:
+                if x != y and "eng_Latn" in [x, y]:
+                    pair = f"{x}-{y}"
+                    hf_lang_subset2isolang[pair] = [
+                        x.replace("_", "-"),
+                        y.replace("_", "-"),
+                    ]
+    return hf_lang_subset2isolang
+
+
+_LANGUAGES_MAPPING = extend_lang_pairs_english_centric()
+
+
+class BibleNLPBitextMining(AbsTaskBitextMining, CrosslingualTask):
+    metadata = TaskMetadata(
+        name="BibleNLPBitextMining",
+        dataset={
+            "path": "davidstap/biblenlp-corpus-mmteb",
+            "revision": "264a18480c529d9e922483839b4b9758e690b762",
+            "split": f"train[:{_N}]",
+            "trust_remote_code": True,
+        },
+        description="Partial Bible translations in 829 languages, aligned by verse.",
+        reference="https://arxiv.org/abs/2304.09919",
+        type="BitextMining",
+        category="s2s",
+        eval_splits=_SPLIT,
+        eval_langs=_LANGUAGES_MAPPING,
+        main_score="f1",
+        date=("2023-04-09", "2023-04-09"),
+        form=["written"],
+        domains=["Religious"],
+        task_subtypes=[],
+        license="CC-BY-SA-4.0",
+        socioeconomic_status="medium",
+        annotations_creators="expert-annotated",
+        dialect=[],
+        text_creation="created",
+        n_samples={"train": _N},
+        avg_character_length={"train": 120},
+        bibtex_citation="""
+@article{akerman2023ebible,
+  title={The eBible Corpus: Data and Model Benchmarks for Bible Translation for Low-Resource Languages},
+  author={Akerman, Vesa and Baines, David and Daspit, Damien and Hermjakob, Ulf and Jang, Taeho and Leong, Colin and Martin, Michael and Mathew, Joel and Robie, Jonathan and Schwarting, Marcus},
+  journal={arXiv preprint arXiv:2304.09919},
+  year={2023}
+}
+""",
+    )
+
+    def load_data(self, **kwargs: Any) -> None:
+        """Load dataset from HuggingFace hub"""
+        if self.data_loaded:
+            return
+
+        self.dataset = {}
+        seen_pairs = []
+        for lang in self.langs:
+            # e.g. 'aai_Latn-eng_Latn' -> 'eng-aai'
+            hf_lang_name = self._transform_lang_name_hf(lang)
+
+            if hf_lang_name in seen_pairs:
+                self.dataset[lang] = self.dataset[self._swap_substrings(lang)]
+            else:
+                dataset = datasets.load_dataset(
+                    name=self._transform_lang_name_hf(lang),
+                    **self.metadata_dict["dataset"],
+                )
+                self.dataset[lang] = datasets.DatasetDict({"train": dataset})
+                seen_pairs.append(hf_lang_name)
+
+        self.dataset_transform()
+        self.data_loaded = True
+
+    def dataset_transform(self) -> None:
+        # Convert to standard format
+        for lang in self.langs:
```

Comment:
if we edit load data already, I don't see why we would need to iterate over `self.langs` twice? maybe we don't need `dataset_transform` at all?

Reply:
Sure we can do the transformation in `load_data` to prevent iterating twice, I'll change it.

Reply:
On second thought: This would not be an improvement because it requires different transformations for if else. (The if else is there to prevent double loading of e.g. 'aai_Latn-eng_Latn' and 'eng_Latn-aai_Latn', which is the same huggingface data, and loading data is slow #572).

IMO the current code is a better trade-off between readability and efficiency. The impact of the second loop is negligible; it is a very cheap operation over 829*2 items.